lib/core/viewer/components/img-viewer.component.spec.ts

```diff
@@ -344,11 +148,147 @@ describe('Test Img viewer component ', () => {
+            expect(rotateButtonElement).not.toEqual(null);
+        });
+
+        it('should not show rotate button by default', () => {
+            const rotateButtonElement = element.querySelector('#viewer-rotate-button');
+            expect(rotateButtonElement).toEqual(null);
+        });
+
+        it('should rotate image by -90 degrees on button click', fakeAsync(() => {
+            component.readOnly = false;
+            spyOn(component, 'rotateImage').and.callThrough();
+            spyOn(component.cropper, 'rotate');
+            fixture.detectChanges();
+            const rotateButtonElement = fixture.debugElement.query(By.css('#viewer-rotate-button'));
+            rotateButtonElement.triggerEventHandler('click', null);
+            tick();
+            expect(component.rotateImage).toHaveBeenCalled();
+            component.rotateImage();
+            expect(component.cropper.rotate).toHaveBeenCalledWith(-90);
+        }));
+
+        it('should display or not the second toolbar if in read only mode', fakeAsync(() => {
+            component.isEditing = true;
+            component.readOnly = true;
+            fixture.detectChanges();
+            let secondaryToolbar = document.querySelector('.adf-secondary-toolbar');
+            expect(secondaryToolbar).toEqual(null);
+
+            component.readOnly = false;
+            fixture.detectChanges();
+            secondaryToolbar = document.querySelector('.adf-secondary-toolbar');
+            expect(secondaryToolbar).not.toEqual(null);
+        }));
+
+        it('should display second toolbar in rotate mode', fakeAsync(() => {
+            component.readOnly = false;
+            expect(component.isEditing).toEqual(false);
+            component.rotateImage();
+            expect(component.isEditing).toEqual(true);
+            fixture.detectChanges();
+            const secondaryToolbar = document.querySelector('.adf-secondary-toolbar');
+            const resetButton = document.querySelector('#viewer-cancel-button');
+            const saveButton = document.querySelector('#viewer-save-button');
+            expect(secondaryToolbar).not.toEqual(null);
+            expect(resetButton).not.toEqual(null);
+            expect(saveButton).not.toEqual(null);
+        }));
+
+        it('should reset the scale and hide second toolbar', fakeAsync(() => {
+            component.readOnly = false;
+            component.isEditing = true;
+            spyOn(component, 'reset').and.callThrough();
+            spyOn(component.cropper, 'reset');
+            spyOn(component.cropper, 'zoomTo');
+            fixture.detectChanges();
+            const cancelButtonElement = fixture.debugElement.query(By.css('#viewer-cancel-button'));
+            cancelButtonElement.triggerEventHandler('click', null);
+            tick();
+            expect(component.reset).toHaveBeenCalled();
+            component.reset();
+            expect(component.scale).toEqual(1.0);
```

Comment:
Can we have the preconditions and expectations in a different block by leaving an empty line? Same for all the other occurrences

Style 05-14
https://alfresco.atlassian.net/wiki/spaces/AP/pages/471060/Code+Style+E2E

Reply:
See previous comment: https://github.com/Alfresco/alfresco-ng2-components/pull/6958#discussion_r623893715